src/osd/PrimaryLogPG.cc

```diff
@@ -584,6 +585,8 @@ bool PrimaryLogPG::should_send_op(
   }
   if (async_recovery_targets.count(peer) && peer_missing[peer].is_missing(hoid)) {
     should_send = false;
+    missing_loc.add_location(hoid, peer);
+    missing_loc.add_missing(hoid, v, eversion_t());
```

Comment:
putting these missing_loc updates in issue_repop() along with the peer_missing updates, before the submit_transaction() call there, would make it easier to find - should_send_op() doesn't sound like it should be modifying state